File: ALDS/1_A.js
'use strict';

function Main(input) {
	input = input.split('\n');
	const N = Number(input[0]);
	const A = input[1].split(' ').map(Number);
	insertionSort(A, N);
}

function insertionSort(A, N) {
	for (let i = 1; i < N; i++) {
		console.log(A.join(' '));
		const v = A[i];
		let j = i - 1;
		while (j >= 0 && A[j] > v) {
			A[j + 1] = A[j];
			j--;
		}
		A[j + 1] = v;
	}
	console.log(A.join(' '));
}

Main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: ALDS1/1_A.py
N = int(input())
A = list(map(int, input().split()))

for i in range(1, N):
    print(" ".join([str(a) for a in A]))
    v = A[i]
    j = i - 1
    while j >= 0 and A[j] > v:
        A[j + 1] = A[j]
        j -= 1
    A[j + 1] = v
print(" ".join([str(a) for a in A]))